<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <arg name="use_nodelet" default="true"/>
  <arg name="manager_name" default="nodelet_manager"/>
  <arg name="input_points_topic" default="points_in"/>
  <arg name="input_feature_points_topic" default="feature_points_in"/>
  <arg name="odom_frame" default="odom"/>
  <arg name="map_frame" default="map"/>
  <arg name="translation_threshold" default="0.1"/>
  <arg name="rotation_threshold" default="0.1"/>
  <arg name="voxel_grid_leaf_size" default="0.1"/>

  <group if="$(arg use_nodelet)">
    <node pkg="nodelet" type="nodelet" name="$(arg manager_name)" args="manager" output="screen"/>

    <node pkg="nodelet" type="nodelet" name="optimization_loader" output="screen" args="load line_rot_slam/MapOptimizationNodelet $(arg manager_name)">
      <param name="input_points_topic" value="$(arg input_points_topic)"/>
      <param name="input_feature_points_topic" value="$(arg input_feature_points_topic)"/>
      <param name="odom_frame" value="$(arg odom_frame)"/>
      <param name="map_frame" value="$(arg map_frame)"/>
      <param name="translation_threshold" value="$(arg translation_threshold)"/>
      <param name="rotation_threshold" value="$(arg rotation_threshold)"/>
      <param name="voxel_grid_leaf_size" value="$(arg voxel_grid_leaf_size)"/>
    </node>
  </group>

  <group unless="$(arg use_nodelet)">
    <node pkg="nodelet" type="nodelet" name="optimization_loader" output="screen" args="standalone line_rot_slam/MapOptimizationNodelet">
      <param name="input_points_topic" value="$(arg input_points_topic)"/>
      <param name="input_feature_points_topic" value="$(arg input_feature_points_topic)"/>
      <param name="odom_frame" value="$(arg odom_frame)"/>
      <param name="map_frame" value="$(arg map_frame)"/>
      <param name="translation_threshold" value="$(arg translation_threshold)"/>
      <param name="rotation_threshold" value="$(arg rotation_threshold)"/>
      <param name="voxel_grid_leaf_size" value="$(arg voxel_grid_leaf_size)"/>
    </node>
  </group>
</launch>
